### ISSUE TITLE
[6.1->7.0] error when running the script due to new api adaptation

### ISSUE THREAD

**Issue by bodi000** [OCA/openupgradelib#95]
Hello,

This appears to be a regression from commit 22a6409, as I did a few upgrades last year without the error. 
The line
https://github.com/OCA/openupgradelib/blob/9b5d8f9710b19339438fcd62e3f98c631086a8cf/openupgradelib/openupgrade.py#L811

produces this error:

```
2018-02-11 10:27:31,701 15701 INFO bs_05_migrated OpenUpgrade: base: post-migration script called with version 6.1.1.3
2018-02-11 10:27:31,701 15701 ERROR bs_05_migrated OpenUpgrade: base: error in migration script base/migrations/7.0.1.3/post-migration.py: TypeError('get() takes exactly 2 arguments (3 given)',)
2018-02-11 10:27:31,702 15701 ERROR bs_05_migrated OpenUpgrade: get() takes exactly 2 arguments (3 given)
Traceback (most recent call last):
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 1391, in wrapped_function
    if use_env2 else cr, version)
  File "/var/tmp/openupgrade/7.0/server/openerp/addons/base/migrations/7.0.1.3/post-migration.py", line 514, in migrate
    openupgrade.set_defaults(cr, pool, force_defaults, force=True)
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 811, in set_defaults
    obj = pool.get(model, False)
TypeError: get() takes exactly 2 arguments (3 given)
2018-02-11 10:27:31,702 15701 ERROR bs_05_migrated openerp.modules.loading: Error executing post migration script for module base
: get() takes exactly 2 arguments (3 given)


2018-02-11 10:27:31,718 15701 ERROR bs_05_migrated openerp: Failed to initialize database `bs_05_migrated`.
Traceback (most recent call last):
  File "/var/tmp/openupgrade/7.0/server/openerp/cli/server.py", line 97, in preload_registry
    db, registry = openerp.pooler.get_db_and_pool(dbname,update_module=update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/pooler.py", line 33, in get_db_and_pool
    registry = RegistryManager.get(db_name, force_demo, status, update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/registry.py", line 203, in get
    update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/registry.py", line 233, in new
    openerp.modules.load_modules(registry.db, force_demo, status, update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/loading.py", line 329, in load_modules
    loaded_modules, processed_modules = load_module_graph(cr, graph, status, perform_checks=update_module, report=report, registry=registry)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/loading.py", line 231, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/migration.py", line 191, in migrate_module
    mod.migrate(self.cr, pkg.installed_version)
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 1391, in wrapped_function
    if use_env2 else cr, version)
  File "/var/tmp/openupgrade/7.0/server/openerp/addons/base/migrations/7.0.1.3/post-migration.py", line 514, in migrate
    openupgrade.set_defaults(cr, pool, force_defaults, force=True)
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 811, in set_defaults
    obj = pool.get(model, False)
TypeError: get() takes exactly 2 arguments (3 given)
```
Is there something I could do to circumvent this situation, as I see no branch per version, and I have updated the old script. Of course, I can go and undo the changes, but this is not an elegant way...

If I simply take out the ```, False``` I get into a new error few lines down, as the new api is called...

Thank you in advance

**Comment by bodi000** [comment]
This is the error for the api call:
```
2018-02-11 10:59:35,624 1832 INFO bs_05_migrated openerp.modules.migration: module base: Running migration [7.0.1.3>] post-migration
2018-02-11 10:59:35,627 1832 INFO bs_05_migrated OpenUpgrade: base: post-migration script called with version 6.1.1.3
2018-02-11 10:59:35,627 1832 ERROR bs_05_migrated OpenUpgrade: base: error in migration script base/migrations/7.0.1.3/post-migration.py: NameError("global name 'api' is not defined",)
2018-02-11 10:59:35,628 1832 ERROR bs_05_migrated OpenUpgrade: global name 'api' is not defined
Traceback (most recent call last):
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 1391, in wrapped_function
    if use_env2 else cr, version)
  File "/var/tmp/openupgrade/7.0/server/openerp/addons/base/migrations/7.0.1.3/post-migration.py", line 514, in migrate
    openupgrade.set_defaults(cr, pool, force_defaults, force=True)
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 817, in set_defaults
    if isinstance(pool, api.Environment):
NameError: global name 'api' is not defined
2018-02-11 10:59:35,629 1832 ERROR bs_05_migrated openerp.modules.loading: Error executing post migration script for module base
: global name 'api' is not defined
2018-02-11 10:59:35,644 1832 ERROR bs_05_migrated openerp: Failed to initialize database `bs_05_migrated`.
Traceback (most recent call last):
  File "/var/tmp/openupgrade/7.0/server/openerp/cli/server.py", line 97, in preload_registry
    db, registry = openerp.pooler.get_db_and_pool(dbname,update_module=update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/pooler.py", line 33, in get_db_and_pool
    registry = RegistryManager.get(db_name, force_demo, status, update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/registry.py", line 203, in get
    update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/registry.py", line 233, in new
    openerp.modules.load_modules(registry.db, force_demo, status, update_module)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/loading.py", line 329, in load_modules
    loaded_modules, processed_modules = load_module_graph(cr, graph, status, perform_checks=update_module, report=report, registry=registry)
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/loading.py", line 231, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/var/tmp/openupgrade/7.0/server/openerp/modules/migration.py", line 191, in migrate_module
    mod.migrate(self.cr, pkg.installed_version)
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 1391, in wrapped_function
    if use_env2 else cr, version)
  File "/var/tmp/openupgrade/7.0/server/openerp/addons/base/migrations/7.0.1.3/post-migration.py", line 514, in migrate
    openupgrade.set_defaults(cr, pool, force_defaults, force=True)
  File "/var/tmp/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 817, in set_defaults
    if isinstance(pool, api.Environment):
NameError: global name 'api' is not defined
```

**Comment by pedrobaeza** [comment]
Fixed with #99
